certification/internal/policy/operator/scc_check.go

```diff
@@ -0,0 +1,121 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/bundle"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+)
+
+// SecurityContextConstraintsCheck evaluates the csv and logs a message if a non default security context constraint is
+// needed by the operator
+
+type SecurityContextConstraintsCheck struct {
+	customSCCSpecified  bool
+	builtInSCCSpecified bool
+}
+
+func getDefaultSCCs() []string {
+	return []string{"restricted", "privileged", "nonroot", "node-exporter", "hostnetwork", "hostmount-anyuid", "hostaccess", "anyuid"}
+}
+
+func NewSecurityContextConstraintsCheck() *SecurityContextConstraintsCheck {
+	return &SecurityContextConstraintsCheck{
+		false,
+		false,
+	}
+}
+
+func (p *SecurityContextConstraintsCheck) Validate(ctx context.Context, bundleRef certification.ImageReference) (bool, error) {
+	requestedSCCList, err := p.dataToValidate(ctx, bundleRef.ImageFSPath)
+	if err != nil {
+		return false, err
+	}
+
+	return p.validate(ctx, requestedSCCList)
+}
+
+func (p *SecurityContextConstraintsCheck) dataToValidate(ctx context.Context, imagePath string) ([]string, error) {
+	csvFilepath, err := bundle.GetCsvFilePathFromBundle(imagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	csvFileReader, err := os.Open(csvFilepath)
+	if err != nil {
+		return nil, err
+	}
+
+	requestedSccList, err := bundle.GetSecurityContextConstraints(ctx, csvFileReader)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract security context constraints from ClusterServiceVersion: %w", err)
+	}
+
+	return requestedSccList, nil
+}
+
+func (p *SecurityContextConstraintsCheck) validate(ctx context.Context, requestedSccList []string) (bool, error) {
+	defaultSccList := getDefaultSCCs()
+
+	if requestedSccList == nil || len(requestedSccList) == 0 {
```

Comment:
```suggestion
	if len(requestedSccList) == 0 {
```

Just check the length. It should be 0 if `requestedSccList` is nil. I'd rather we returned `[]Type{}` from the generating function, but either way, this should be sufficient.

https://go.dev/play/p/_WAM2hd1YDL

Reply:
+1

Reply:
will fix